packages/app/index.ts

```diff
@@ -23,7 +23,12 @@ import { Logger } from '@firebase/logger';
 const logger = new Logger('@firebase/app');
 
 // Firebase Lite detection
-if (isBrowser() && 'firebase' in self) {
+if (
+  isBrowser() &&
+  'firebase' in self &&
+  // Check that firebase is not undefined
+  ((self as any).firebase as FirebaseNamespace) !== undefined
```

Comment:
`'firebase' in self` becomes redundant, so we can remove it.

Can you add `// elsint-disable-next-line @typescript-eslint/no-explicit-any` on the top of the line to pass lint ?
Also `as FirebaseNamespace` is not necessary, so it can be dropped.